index.js

```diff
@@ -1,39 +1,256 @@
+// I was trying to create a solution using a state machine but couldn't get it to work -- so I scraped it and followed a similar format to your solutions... 
+
 const readline = require('readline');
-const readlineInterface = readline.createInterface(process.stdin, process.stdout);
+
+const readlineInterface = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     readlineInterface.question(questionText, resolve);
   });
+};
+
+//--------------------------
+
+//player
+const player = {
+  currentScene: null,
+  inventory: [],
+  status: [],
+
+  //move
+  changeScene: (room) => {
+    player.currentScene = room
+  },
+
+  //pick up
+  pickUp: (item) => {
+    if (item.takeable === true) {
+      player.inventory.push(item);
+      player.currentScene.inventory.pop(item)
+      return `You pick up a ${item.name}`
+    } else {
+      return "You can't take that"
+    }
+  },
+
+  //use items
+  useItem: (item) => {
+    item.action()
+  }
 }
 
-// remember the StateMachine lecture
-// https://bootcamp.burlingtoncodeacademy.com/lessons/cs/state-machines
-let states = {
-  'roomOne': { canChangeTo: [ 'roomTwo' ] },
-  'roomTwo': { canChangeTo: [ 'roomThree' ] },
-  'roomThree': { canChangeTo: [ 'roomOne' ] }
-};
+//Scene Constructor
+class Scene {
+  constructor(name, description, inventory, forward, backward, right, left) {
+
+    this.name = name;
+    this.description = description;
+    this.inventory = inventory || [];
+    this.isLocked = false;
+    this.forward = forward || null;
+    this.backward = backward || null;
+    this.right = right || null;
+    this.left = left || null;
+
+    this.takeItem = (item) => {
+      this.inventory.pop(item)
+    };
```

Comment:
Popping off an array will actually only remove the last item in an array, even if you pass it an argument. This works fine if you only have one object in the inventory, but when you have multiple items you might not remove the item you think you're removing.  This was an error in my original code that it took me a while to figure out.  To remove an item from an array you need the index of the item, which you can get by using the `.find((item) => item.name === stash)` method on the array, and then you can use the `.splice(index, num)` method to remove an item by passing it the index, and the number of items to be removed (1 in this case)